src/mlpack/core/data/bow_encoding.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file bow.hpp
+ * @author Jeffin Sam
+ *
+ * Definition of Bag of Words encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_BOW_ENCODING_HPP
+#define MLPACK_CORE_DATA_BOW_ENCODING_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+#include <utility>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple Bow Enocding class
+ */
+class Bow
+{
+ public:
+  /**
+  * A function to create mapping from a given corpus.
+  * 
+  * @param input Corpus of text to encode.
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void CreateMap(std::string& input, TokenizerType tokenizer);
+  /*
+  * Default Constructor
+  */
+  Bow() {}
+  /*
+  * Copy Constructor.
+  */
+  Bow(const Bow& oldObject);
+  /*
+  * Move Constructor.
+  */
+  Bow(Bow&& oldObject) = default;
+  /*
+  * Move Assignment Operator.
+  */  
+  Bow& operator= (Bow&& oldObject) = default;
+  /*
+  * Assignment Operator.
+  */
+  void operator= (const Bow& oldObject);
```

Comment:
I guess the return value should be `Bow&`.

Reply:
Umm, not sure, but i did make the change :)

Reply:
It is possible to write
```
int a, b;
a = b = 0;
```